synapse/handlers/oidc_handler.py

```diff
@@ -869,73 +865,50 @@ async def _map_userinfo_to_user(
         # to be strings.
         remote_user_id = str(remote_user_id)
 
-        # first of all, check if we already have a mapping for this user
-        previously_registered_user_id = await self._sso_handler.get_sso_user_by_remote_user_id(
-            self._auth_provider_id, remote_user_id,
+        # Older mapping providers don't accept the `failures` argument, so we
+        # try and detect support.
+        mapper_args = inspect.getfullargspec(
+            self._user_mapping_provider.map_user_attributes
         )
-        if previously_registered_user_id:
-            return previously_registered_user_id
+        supports_failures = "failures" in mapper_args.args
```

Comment:
We also could potentially just doc this, but it seemed easy enough to support both.

Reply:
Keeping backwards compat seems like the easiest way forwards